File: packages/python-sdk/octopoid_sdk/examples/auto_approve.py
#!/usr/bin/env python3
"""
Example: Auto-approve low-risk tasks

This script automatically approves tasks that meet certain criteria:
- Documentation tasks with commits
- Test tasks with commits

Usage:
    OCTOPOID_SERVER_URL=https://... python auto_approve.py
"""

from octopoid_sdk import OctopoidSDK


def main():
    sdk = OctopoidSDK()

    print("🔍 Checking for tasks to auto-approve...\n")

    # Get provisional tasks (awaiting review)
    tasks = sdk.tasks.list(queue="provisional")

    if not tasks:
        print("No tasks awaiting review")
        return

    print(f"Found {len(tasks)} task(s) awaiting review\n")

    approved = 0

    for task in tasks:
        # Auto-approve docs and test tasks with commits
        should_auto_approve = (
            task.get("role") in ("docs", "test")
            and task.get("commits_count", 0) > 0
        )

        if should_auto_approve:
            print(f"✅ Auto-approving {task['role']} task: {task['id']}")

            sdk.tasks.accept(task["id"], accepted_by="auto-approve-script")

            approved += 1
        else:
            role = task.get("role") or "unknown"
            print(f"⏭️  Skipping {role} task: {task['id']}")

    print(f"\n✅ Auto-approved {approved} task(s)")


if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"❌ Error: {e}")
        exit(1)

File: packages/sdk/examples/auto-approve.ts
#!/usr/bin/env node
/**
 * Example: Auto-approve low-risk tasks
 *
 * This script automatically approves tasks that meet certain criteria:
 * - Documentation tasks with commits
 * - Test tasks with commits
 *
 * Usage:
 *   OCTOPOID_SERVER_URL=https://... npm run example:auto-approve
 */

import { OctopoidSDK } from '../src/index'

async function main() {
  const sdk = new OctopoidSDK()

  console.log('🔍 Checking for tasks to auto-approve...\n')

  // Get provisional tasks (awaiting review)
  const tasks = await sdk.tasks.list({ queue: 'provisional' })

  if (tasks.length === 0) {
    console.log('No tasks awaiting review')
    return
  }

  console.log(`Found ${tasks.length} task(s) awaiting review\n`)

  let approved = 0

  for (const task of tasks) {
    const shouldAutoApprove =
      // Auto-approve docs tasks with commits
      (task.role === 'docs' && (task.commits_count || 0) > 0) ||
      // Auto-approve test tasks with commits
      (task.role === 'test' && (task.commits_count || 0) > 0)

    if (shouldAutoApprove) {
      console.log(`✅ Auto-approving ${task.role} task: ${task.id}`)

      await sdk.tasks.accept(task.id, {
        accepted_by: 'auto-approve-script',
      })

      approved++
    } else {
      console.log(`⏭️  Skipping ${task.role || 'unknown'} task: ${task.id}`)
    }
  }

  console.log(`\n✅ Auto-approved ${approved} task(s)`)
}

main().catch((error) => {
  console.error('❌ Error:', error.message)
  process.exit(1)
})
